sonar-php-plugin/src/main/java/org/sonar/plugins/php/PHPSensor.java

```diff
@@ -133,14 +138,51 @@ public void analyse(Project project, SensorContext context) {
     progressReport.start(Lists.newArrayList(fileSystem.files(mainFilePredicate)));
 
     for (InputFile inputFile : inputFiles) {
-      saveIssues(phpAnalyzer.analyze(inputFile.file()), inputFile);
       progressReport.nextFile();
+      phpAnalyzer.nextFile(inputFile.file());
+
+      saveIssues(phpAnalyzer.analyze(), inputFile);
+      saveHighlighting(phpAnalyzer.getHighlighting(), inputFile);
+      saveNewFileMeasures(phpAnalyzer.computeMeasures(fileLinesContextFactory.createFor(inputFile)), inputFile);
     }
 
     progressReport.stop();
 
   }
 
+  private void saveHighlighting(List<HighlightingData> highlightingDataList, InputFile inputFile) {
+    Highlightable highlightable = resourcePerspectives.as(Highlightable.class, inputFile);
+
+    if (highlightable == null) {
+      LOG.warn("Could not get " + Highlightable.class.getCanonicalName() + " for " + inputFile.file());
+
+    } else {
+      HighlightingBuilder highlightingBuilder = highlightable.newHighlighting();
+      for (HighlightingData highlightingData : highlightingDataList) {
+        highlightingBuilder.highlight(highlightingData.startOffset(), highlightingData.endOffset(), highlightingData.highlightCode());
+      }
+      highlightingBuilder.done();
+    }
+  }
+
+  private void saveNewFileMeasures(FileMeasures fileMeasures, InputFile inputFile) {
+    context.saveMeasure(inputFile, CoreMetrics.LINES, fileMeasures.getLinesNumber());
+    context.saveMeasure(inputFile, CoreMetrics.NCLOC, fileMeasures.getLinesOfCodeNumber());
+//    context.saveMeasure(inputFile, CoreMetrics.COMMENT_LINES, fileMeasures.getCommentLinesNumber());
+    context.saveMeasure(inputFile, CoreMetrics.CLASSES, fileMeasures.getClassNumber());
+    context.saveMeasure(inputFile, CoreMetrics.FUNCTIONS, fileMeasures.getFunctionNumber());
+    context.saveMeasure(inputFile, CoreMetrics.STATEMENTS, fileMeasures.getStatementNumber());
+
+    context.saveMeasure(inputFile, CoreMetrics.COMPLEXITY, fileMeasures.getFileComplexity());
+    context.saveMeasure(inputFile, CoreMetrics.COMPLEXITY_IN_CLASSES, fileMeasures.getClassComplexity());
+    context.saveMeasure(inputFile, CoreMetrics.COMPLEXITY_IN_FUNCTIONS, fileMeasures.getFunctionComplexity());
+//
+//    context.saveMeasure(inputFile, fileMeasures.getFunctionComplexityDistribution().build(true).setPersistenceMode(PersistenceMode.MEMORY));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)